File: app/src/main/java/ua/syt0r/furiganit/model/repository/user/UserRepository.java
package ua.syt0r.furiganit.model.repository.user;

import com.google.firebase.auth.FirebaseAuth;

public class UserRepository {

    private FirebaseAuth firebaseAuth;

    public UserRepository() {
        firebaseAuth = FirebaseAuth.getInstance();
    }

    public void signIn(OnActivityResultListener listener) {

    }

    public void signOut() {
        firebaseAuth.signOut();
    }

    public boolean isLogged() {
        return firebaseAuth.getCurrentUser() != null;
    }

    public interface OnActivityResultListener {
        void onActivityResult();
    }

}

File: app/src/main/java/ua/syt0r/furiganit/model/repository/user/UserRepository.kt
package ua.syt0r.furiganit.model.repository.user

import io.reactivex.Completable
import io.reactivex.Single
import ua.syt0r.furiganit.model.entity.UserData

interface UserRepository {
    fun saveUserData(userData: UserData): Completable
    fun getUserData(): Single<UserData>
    fun clearUserData(): Completable
}
